From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Anderson <thomasanderson@chromium.org>
Date: Wed, 20 Nov 2019 22:02:10 +0000
Subject: X11: Fix color masks for ColorTypeForVisual()

BUG=1025266
R=rjkroege

Change-Id: Id1e67e64e1eb433d26a9f54c1d2b201c8215c71b
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1922697
Commit-Queue: Thomas Anderson <thomasanderson@chromium.org>
Auto-Submit: Thomas Anderson <thomasanderson@chromium.org>
Reviewed-by: Robert Kroeger <rjkroege@chromium.org>
Cr-Commit-Position: refs/heads/master@{#717251}

diff --git a/ui/base/x/x11_util.cc b/ui/base/x/x11_util.cc
index 479ef7177f888f74406abe354a181af5ea79046a..61e3c9c03296b6b09166dca067d20534cc7725b3 100644
--- a/ui/base/x/x11_util.cc
+++ b/ui/base/x/x11_util.cc
@@ -1295,8 +1295,8 @@ SkColorType ColorTypeForVisual(void* visual) {
     unsigned long green_mask;
     unsigned long blue_mask;
   } color_infos[] = {
-      {kRGB_565_SkColorType, 0x1f, 0x7e0, 0xf800},
-      {kARGB_4444_SkColorType, 0xf0, 0xf00, 0xf000},
+      {kRGB_565_SkColorType, 0xf800, 0x7e0, 0x1f},
+      {kARGB_4444_SkColorType, 0xf000, 0xf00, 0xf0},
       {kRGBA_8888_SkColorType, 0xff, 0xff00, 0xff0000},
       {kBGRA_8888_SkColorType, 0xff0000, 0xff00, 0xff},
       {kRGBA_1010102_SkColorType, 0x3ff, 0xffc00, 0x3ff00000},
@@ -1309,7 +1309,10 @@ SkColorType ColorTypeForVisual(void* visual) {
       return color_info.color_type;
     }
   }
-  LOG(FATAL) << "Unsupported visual: " << XVisualIDFromVisual(vis);
+  LOG(FATAL) << "Unsupported visual with rgb mask 0x" << std::hex
+             << vis->red_mask << ", 0x" << vis->green_mask << ", 0x"
+             << vis->blue_mask
+             << ".  Please report this to https://crbug.com/1025266";
   return kUnknown_SkColorType;
 }
 
